### DISCUSSION TITLE
How to load Python-plugin sub-modules in Far2m

### DISCUSSION THREAD

**Discussion by sorcodiv** [shmuz/far2m#72] [Q&A] [answered]
Здравствуйте!

Столкнулся с вопросом, есть ли возможность в Far2m загружать модули к Python-плагину?
Я собрал Far2m с Python-плагином (чуть подробнее о сборке тут: [How to build and run far2l python plugin on ubuntu 16.04](https://github.com/sorcodiv/far2lua-build/wiki/How-to-build-and-run-Far2l-Python-plugin-on-Ubuntu-16.04)), плагин загружается (виден в lm:farabout), но при попытке загрузить суб-плагин отправляет команду в консоль:

```
$ py:load ucharmap
py:load: command not found
```

В Far2l - команда работает ожидаемо (т.е. загружает модуль ucharmap.py)

в lm:farabout видно:
```
╔═══════════════════════════════════ About far2m ═════╗
║  FAR2M version    │ 2.6.0-2024-07-30-9ff14a5-beta   ║
║  Platform         │ x64                             ║
...
║  -- Plugins (18)  │                                 ║
...
║  Python           │ Python plugin                   ║
```

Одна из целей сборки Python-плагина - воспользоваться аналогом [FARdroid](https://forum.farmanager.com/viewtopic.php?t=5334), который кратко упоминается в [доке](https://github.com/akruphi/far2l/wiki):
+ про python-плагин **uadb** для доступа к андроиду из фара: https://github.com/elfmz/far2l/issues/1207#issuecomment-1523735293

Можно ли в Far2m добавить вызов плагина py:load (может быть, каким-то образом через lua-враппер)?

P.S. Спасибо за [menu items GUID's](https://github.com/shmuz/far2m/commit/cd4a44d74b61881b361fb9d8ef5676d575903035), и за lm:farabout - отдельное спасибо :)

**Comment by shmuz** [comment]
Посмотрел, есть такая неисправность.
Сломалось 12 января, но я не замечал, потому что постоянно собираю без этого плагина.
Надо будет разобраться.

**Comment by shmuz** [comment, marked as answer]
Можно проверять.

**Comment by shmuz** [comment]
@sorcodiv Вы проверили?

**Comment by sorcodiv** [comment]
Да, проверил, на примере ucharmap - загружается и работает, спасибо! Даже удалось py-плагин заставить работать в режиме "portable" (с оговорками), но размер плагина+python более чем в два раза превышает остальной far со всеми необходимыми библиотеками. Пока не стал выкладывать такую сборку, попробую поискать пути уменьшения размера.
